### PR TITLE
TYLERB/APPEALS-16329: Create Reusable Issue Type Column

### PR DESCRIPTION
**Pull request by TylerBroyles**
Resolves [APPEALS-16329](https://vajira.max.gov/browse/APPEALS-16329)
Also Resolves [APPEALS-16340](https://vajira.max.gov/browse/APPEALS-16340)
Also Resolves [APPEALS-16538](https://vajira.max.gov/browse/APPEALS-16538)
Also Resolves [APPEALS-17393](https://vajira.max.gov/browse/APPEALS-17393)
Also Resolves [APPEALS-16696](https://vajira.max.gov/browse/APPEALS-16696)

### Description
Created an issue type column and added it to the decision review queue.

1. Created an Issue Type column to be used with the Queue Table component. 
2. Added an attribute for issue_types to the decision_review_task_serializer for use in the decision review queues. 
3. Added this new column to the decision review queues in both the in progress tasks tab and the completed tasks tab.
4. Added an alias to the business line model to sort, display, and serialize data for this new column.
5. Automatically sort the alias alphabetically if there are more than 1 issue type on the same decision review. 
6. Added the ability to sort by this new column in the decision review queues.

### Acceptance Criteria
- [ ] Reusable Issue Type Column
  - [ ] Created the new issues type column in TaskTableColumns.jsx 
  - [ ] Added the new column definition in the QUEUE_CONFIG.json file
- [ ] Add the issues_types attribute to the decision_review_task_serializer
- [ ] Adding the Issue Type column to the decision review queue
  - [ ] The column is visible in the DR queue and matches the designs
  - [ ] If there are more than 1 issue category, the issue types are sorted automatically sorted by alphabetical order on the appeal
  - [ ] If there are multiple issues and they have the same issue category, that issue category should only display once in the issue_type column e.g if there are two issues with the type "Caregiver | Other" it should only display "Caregiver | Other" once 
  - [ ] The column is sortable in the DR queue
- [ ] All of the above for the In Progress Tab
- [ ] All of the above for the completed tab 

### Testing Plan
1. Setup data
```ruby
50.times do
  vha_task = FactoryBot.create(:higher_level_review_vha_task)
  vha_task.appeal.request_issues = []
  num_objects = rand(1..4)
  num_objects.times do
	vha_task.appeal.request_issues << FactoryBot.create(:request_issue, benefit_type: 'vha', nonrating_issue_category: Constants.ISSUE_CATEGORIES.vha.sample, nonrating_issue_description: "VHA - Category")
  end
  vha_task.save
  vha_task.appeal.save
end

50.times do
  vha_task = FactoryBot.create(:supplemental_claim_vha_task)
  vha_task.appeal.request_issues = []
  num_objects = rand(1..4)
  num_objects.times do
	vha_task.appeal.request_issues << FactoryBot.create(:request_issue, benefit_type: 'vha', nonrating_issue_category: Constants.ISSUE_CATEGORIES.vha.sample, nonrating_issue_description: "VHA - Category")
  end
  vha_task.save
  vha_task.appeal.save
end

# Can also create a vha user if you don't have one available
vha_user = create(:user, css_id: "ISSUE_TESTER", full_name: "ISSUE TEST")
BusinessLine.find_by(url: 'vha').add_user(vha_user)
vha_user.save

# Tasks can be completed using the task.completed! method
# For example to complete all the tasks in the vha business line
vha = BusinessLine.find_by(url: 'vha')
vha.tasks.each { |t| t.completed! }
```
2. Login as a VHA business line user
3. Navigate to the decision review queue at the url /decision_reviews/vha
4. Verify that the Issue Type column header is present
5. Verify that the Issue Type data is present in the column
image

6. Verify that the Issue types if there are more than 1 should be sorted alphabetically
image

7. Click on the issue types column header to sort by the issue_types column.
8. Verify it is sorted in ascending order
9. Verify the url search parameters are correct &sort_by=issueTypesColumn&order=asc
image

10. Click on the issue types column header again to sort by the issue_types column
11. Verify it is sorted in descending order
12. Verify the url search parameters are correct &sort_by=issueTypesColumn&order=desc
image
13. Repeat all the steps for the completed tab

### Storybook Story
* [ ] Updated the NonCompTabs.stories.js storybook file

### CODE REVIEW

File: client/app/queue/components/TaskTableColumns.jsx [hunk 1]
```diff
@@ -250,6 +250,35 @@ export const issueCountColumn = (requireDasRecord) => {
...
+    enableFilter: true,
+    filterOptions,
+    columnName: 'appeal.issueTypes',
+    valueFunction: (task) => {
+      const commaDelimitedIssueTypes = task.appeal.issueTypes;
+
+      // Remove duplicates from the comma delimited list of issues
+      const uniqueIssueTypes = [...new Set(commaDelimitedIssueTypes?.split(','))];
+
+      // return uniqueIssueTypes.map((type) => (<span> {type} <br /> </span>));
```
Comment (pamatyatake2) [inline client/app/queue/components/TaskTableColumns.jsx:267]: i think we can remove this commented out code

File: client/app/queue/components/TaskTableColumns.jsx [hunk 2]
```diff
@@ -250,6 +250,35 @@ export const issueCountColumn = (requireDasRecord) => {
...
+
+      // return uniqueIssueTypes.map((type) => (<p key={type}> {type} </p>));
```
Comment (pamatyatake2) [inline client/app/queue/components/TaskTableColumns.jsx:269]: same here-> removing the commented code

**Review by pamatyatake2 — COMMENTED** [review]
looks good.

**Review by ajspotts — APPROVED** [review]
lgtm!

**Pull request merged**